### DISCUSSION TITLE
Started to get randomly  TypeError: can't pickle _thread.RLock objects

### DISCUSSION THREAD

**Discussion by UriSShavit** [behave/behave#1013] [Q&A] [answered]
Hello, 
My team is a heavy user of this great feature.
Recently we started to fail on different test on different steps on TypeError: can't pickle _thread.RLock objects
Has anyone else experienced it?
I ran pip compile on the code and saw that package parse-type==0.6.0 was released this Jan and reverted it back but the error still occurs.
Please help :|

**Comment by spyoungtech** [comment]
Could you provide the full traceback of the error you're seeing?

I'm not 100% sure, but I don't think `behave` depends on `pickle` or marshalling of objects in any way in general, so I'm not confident that `behave` is responsible for the problem you are seeing. The full stack trace would help us make that determination.

**Comment by jenisys** [comment, marked as answer]
`behave` currently uses no threads/threading, neither does `parse_type`.
In addition, `behave` does not make any use of `pickle`.

Therefore, I assume that problem is in your code base or the other dependencies that are used by your code base.
Just search for which module(s) are importing the `pickle` module.

**Comment by UriSShavit** [comment]
I must say I inherited the package from a developer who quit and added the source code to our Artifactory.
I think this might be the reason for the failures.

Here is the traceback, but depending on your answers, it's pointless at this point.
Will update here after re uploading the code to the artifactory:

08:36:01     Then [SF] I expect the results in table amazon_sox.organic_and_paid_conflict_table to be equal to                                                                       # venv/lib/python3.7/site-packages/airflowbigdataoperators/com/kenshoo/shared_tests/component/features/steps/sf_steps.py:68
08:36:01       | KSNAME | CAMPAIGN_ID | ASIN       | KS_AD_ID | ADGROUP_ID | CUST_PROFILE_ID | KEYWORDS       |
08:36:01       | KS1163 | 20702       | B07PZY13MS | 123456   | 27197737   | 1521            | poise 27 count |
08:36:01 
08:36:02 Verifying airflow server is up...
08:36:02 Verifying airflow server is up...
08:36:02 Snowflake query result: 01a381ca-0604-7e71-0000-7041563b1bfe
08:36:02 ##########################
08:36:02 Created the following schema: test_example_schema_b16b307e
08:36:02 ##########################
08:36:02 Snowflake query result: 01a381ca-0604-7e71-0000-7041563b1c0a
08:36:02 Snowflake query result: 01a381ca-0604-8024-0000-7041563b0f2a
08:36:02 Snowflake query result: 01a381cd-0604-8024-0000-7041563b2d7e
08:36:02 ##########################
08:36:02 Created the following schema: amazon_sox_35c2cbc9
08:36:02 ##########################
08:36:02 Snowflake query result: 01a381cd-0604-7e71-0000-7041563b3ace
08:36:02 ##########################
08:36:02 Created the following schema: dwh_prod_35c2cbc9
08:36:02 ##########################
08:36:02 Snowflake query result: 01a381cd-0604-7e71-0000-7041563b3ae2
08:36:02 ##########################
08:36:02 Created the following schema: bi_dwh_prod_35c2cbc9
08:36:02 ##########################
08:36:02 Snowflake query result: 01a381cd-0604-8024-0000-7041563b2dde
08:36:02 Snowflake query result: 01a381cd-0604-7e71-0000-7041563b3b36
08:36:02 Snowflake query resuTraceback (most recent call last):
08:36:02   File "/usr/lib/python3.7/multiprocessing/queues.py", line 236, in _feed
08:36:02     obj = _ForkingPickler.dumps(obj)
08:36:02   File "/usr/lib/python3.7/multiprocessing/reduction.py", line 51, in dumps
08:36:02     cls(buf, protocol).dump(obj)
08:36:02 TypeError: can't pickle _thread.RLock objects

**Comment by UriSShavit** [comment]
Ok I figured out that my code uses behave-parallel and not behave. I posted in the wrong repo. Sorry and thank you for your time
